### PR TITLE
Suss out scheduling

### PR DESCRIPTION
**Pull request by tlunter**
@nahiluhmot @djosborne

Added a task model that has n schedules and belongs to a device.
Changed schedule to belong to a task instead of a device.
A task will set the state of a device to the designated state whenever one of it's schedules has been passed.

A schedule has a start time and optionally an interval and end time.
An interval > 0 will be periodic, and triggered every time an interval integer is crossed.
An interval <= 0 will be a single shot, and triggered once the start time has been passed.

The new state will have the source set to `scheduled`.

~~**NOTE**: I have not included tests yet.~~

### CODE REVIEW

File: app/cappy/services/task_runner.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+      def perform
+        previous = last_completed
+        # Cutoff for tasks
+        now = Models::TaskComplete.create
+
+        tasks(previous.created_at, now.created_at).each do |task|
+          task.device.states << Models::State.create(
+            state: task.state,
+            source: 'scheduled'
```
Comment (nahiluhmot) [inline app/cappy/services/task_runner.rb]: Maybe make this a constant, `State::SCHEDULED`?

**Comment by tlunter** [comment]
@nahiluhmot @djosborne Updated and good to go!

File: db/migrate/20150228210433_create_task_completed.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class CreateTaskCompleted < ActiveRecord::Migration
```
Comment (nahiluhmot) [inline db/migrate/20150228210433_create_task_completed.rb]: I don't understand what this table is for if it has no members / associations.

Comment (tlunter) [inline db/migrate/20150228210433_create_task_completed.rb]: Note it's use [here](https://github.com/cappity-cappity-capstone/backend/pull/24/files#diff-d645137d0abd94bc9dacfd8e73802f80R12) and [here](https://github.com/cappity-cappity-capstone/backend/pull/24/files#diff-d645137d0abd94bc9dacfd8e73802f80R49).  It acts as DB cutoff for which tasks to run.

**Comment by nahiluhmot** [comment]
:+1:

**Pull request merged**